psm_utils/psm_list.py

```diff
@@ -98,12 +98,7 @@ def __getitem__(self, item) -> PSM | list[PSM]:
             return PSMList(psm_list=self.psm_list[item])
         elif isinstance(item, str):
             # Return PSM property as array across full PSMList
-            try:
-                # Let NumPy coerce dtype (e.g., multidimensional arrays)
-                return np.array([psm[item] for psm in self.psm_list])
-            except ValueError:
-                # If dtype is not consistent, force dtype to be object
-                return np.array([psm[item] for psm in self.psm_list], dtype=object)
+            return np.fromiter([psm[item] for psm in self.psm_list], dtype=object, count=len(self))
```

Comment:
On top of adding count, you could also replace the list comprehension with a generator expression, although there's a bit of a CPU usage vs memory usage trade-off.